src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala

```diff
@@ -0,0 +1,41 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.util.ByteString
+import io.iohk.ethereum.jsonrpc.JsonRpcController.{JsonDecoder, JsonEncoder}
+import io.iohk.ethereum.jsonrpc.Web3Service.{ClientVersionRequest, ClientVersionResponse, Sha3Request, Sha3Response}
+import org.json4s.{DefaultFormats, Formats, JValue}
+import org.json4s.JsonAST.{JArray, JString}
+import org.json4s.JsonDSL._
+import org.spongycastle.util.encoders.Hex
+
+import scala.util.{Failure, Success, Try}
+
+object JsonMethodsImplicits {
+
+  import JsonRpcErrors._
+
+  implicit val formats: Formats = DefaultFormats
+
+  implicit val web3_sha3 = new JsonDecoder[Sha3Request] with JsonEncoder[Sha3Response] {
+    override def decodeJson(params: Option[JArray]): Either[JsonRpcError, Sha3Request] =
+      params match {
+        case Some(JArray((input: JString) :: Nil)) if input.s.startsWith("0x") =>
```

Comment:
I sketched something here (haven't tested it tho): https://github.com/input-output-hk/etc-client/blob/feature/jsonrpcBlockNumberGetTransactionByBlockHashAndIndex/src/main/scala/io/iohk/ethereum/jsonrpc/JsonMethodsImplicits.scala#L53